model_utils/tracker.py

```diff
@@ -3,10 +3,43 @@
 
 from django.core.exceptions import FieldError
 from django.db import models
-from django.db.models.fields.files import FileDescriptor
+from django.db.models.fields.files import FieldFile, FileDescriptor
 from django.db.models.query_utils import DeferredAttribute
 
 
+class LightStateFieldFile(FieldFile):
+    """
+    FieldFile subclass with the only aim to remove the instance from the state.
+
+    The change introduced in Django 3.1 on FieldFile subclasses results in pickling the
+    whole instance for every field tracked.
+    As this is done on the initialization of objects, a simple queryset evaluation on
+    Django 3.1+ can make the app unusable, as CPU and memory usage gets easily
+    multiplied by magnitudes.
+    """
+    def __getstate__(self):
+        """
+        We don't need to deepcopy the instance, so nullify if provided.
+        """
+        state = super().__getstate__()
+        if 'instance' in state:
+            state['instance'] = None
+        return state
+
+
+def lightweight_deepcopy(value):
+    """
+    Use our lightweight class to avoid copying the instance on a FieldFile deepcopy.
+    """
+    if isinstance(value, FieldFile):
+        value = LightStateFieldFile(
```

Comment:
Consider just using `FieldFile(instance=None, field=value.field, name=value.name)` here. Will it be more error-prone? I'm not sure if missing `instance` is really debuggable, because now it disappears only while pickling/unpickling.